conn_linux.go

```diff
@@ -328,22 +328,54 @@ func newError(errno int) error {
 
 var _ socket = &sysSocket{}
 
+// netNSExecutor executes a function in a network namespace
+type netNSExecutor interface {
+	// stop stops the executor
+	stop()
+	// run runs the given function in a network namespace of the executor
+	run(f func())
+}
+
 // A sysSocket is a socket which uses system calls for socket operations.
 type sysSocket struct {
 	mu     sync.RWMutex
 	fd     *os.File
 	closed bool
-	g      *lockedNetNSGoroutine
+	g      netNSExecutor
+}
+
+// currentNetNSExecutor executes functions in the current network namespace
+type currentNetNSExecutor struct{}
```

Comment:
selfNetNSExecutor?
self is commonly used in linux stuff like procfs so maybe it sounds more relevant than 'current'